src/_articles/archive/zones.md

```diff
@@ -21,28 +21,28 @@ obsolete: true
 
 ### Asynchronous dynamic extents
 
-_Written by Florian Loitsch and Kathy Walrath<br>
-March 2014_
-
 This article discusses zone-related APIs in the dart:async library,
-with a focus on the top-level `runZoned()` function.
+with a focus on the top-level [`runZoned()`][]
+and [`runZonedGuarded()`][] functions.
 Before reading this article,
 you should be familiar with the techniques covered in
 [Futures and Error Handling](/guides/libraries/futures-error-handling).
 
-Currently, the most common use of zones is
-to handle errors raised in asynchronously executed code.
+[`runZoned()`]: ({{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/runZoned.html)
+[`runZonedGuarded()`]: ({{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/runZonedGuarded.html)
+
+Developers commonly use zones to handle uncaught errors.
 For example,
 a simple HTTP server
 might use the following code:
```

Comment:
Aha, I see. So my issue with the last suggestion AND the way I rewrote it here is because I feel like making it active makes the statement about _developers_' personal preference more than about zones, so it...reads weird to me. 

I'm not sure if I'm explaining that well (?), but basically I don't know how to make it more zone-centric. What do you think of these?
> Zones handle uncaught errors

or

> Zones are useful for handling uncaught errors

or to stick to the original sentence the most

> The most common use of zones is to handle uncaught errors

Reply:
**reply:** OK, I get the distinction. How about...
```markdown
Dart needs a way to handle uncaught errors. 
Zones provide the most common solution to this problem.
```

Reply:
Being pedantic ... Dart is a programming language. It doesn't need anything. (And it doesn't like being anthropomorphized ;).
Also, `try`/`catch` is the most common solution to catching errors. We haven't defined what an "uncaught error" is, so the reader should assume that it's any error which isn't caught (by a `catch`).

So, maybe:

> Uncaught errors are thrown errors which reach the event loop, 
> or error results of futures that nobody waits for.
> An uncaught error is always reported to the current zone of the code which failed to catch it.
> The default behavior is to terminate a native program, or show error messages in the browser console.
> 
> By running code in a custom zone, with a custom _uncaught error handler_ installed,
> you can intercept and handle the uncaught errors in the way you prefer.

Reply:
**reply:** The GSG covers [anthropomorphism](https://developers.google.com/style/anthropomorphism), but I'd argue "need" is not a human quality per se. Maybe `require` would be better. 

Another important distinction: we can't use code elements as nouns or verbs. See the [GSG](https://developers.google.com/style/code-in-text#grammatical-treatment-of-code-elements) on that point.

That said, I would give a bit more context, even though it is basic CS, because:

1. It helps with flow.
2. Not everyone for whom we write understands all CS concepts.

May I suggest:

```markdown
Good programming practice includes providing ways to handle errors in code.
When code tries to handle an error, this process is called a `throw`. The target of that `throw`
would be the process that handles the error which is called a `catch`.
If a `throw` lacks a related `catch`, the result is called an _uncaught error_.
Uncaught errors can also occur when a future completes with an error and no
means exist to wait for that future.

An uncaught error reports to the current zone of the code that failed to catch it.
By default, the program would stop or display the error message in a console.

If you run your code within a custom zone with its own uncaught error handler,
you can intercept uncaught errors and choose how to handle them.
```